cdap-app-fabric/src/main/java/io/cdap/cdap/metadata/profile/ProfileMetadataMessageProcessor.java

```diff
@@ -365,4 +377,41 @@ private void emitApplicationCountMetric() {
       LOG.warn("Failed to get application count", e);
     }
   }
+
+  private void emitPluginCountMetric() {
+    emitPluginCountMetric(null);
+  }
+
+  private void emitPluginCountMetric(Set<ImmutablePair<String, String>> deletedPlugins) {
+    try {
+      List<ImmutablePair<String, String>> remainingPlugins = appMetadataStore.getAllApplications().stream()
+        .map(ApplicationMeta::getSpec)
+        .map(ApplicationSpecification::getPlugins)
+        .flatMap(e -> e.values().stream())
+        .map(Plugin::getPluginClass)
+        .map(e -> ImmutablePair.of(e.getName(), e.getType()))
+        .collect(Collectors.toList());
+
+      Map<ImmutablePair<String, String>, Long> pluginCounts = remainingPlugins.stream()
+        .collect(Collectors.groupingByConcurrent(Function.identity(), Collectors.counting()));
+      for (Map.Entry<ImmutablePair<String, String>, Long> entry : pluginCounts.entrySet()) {
+        Map<String, String> tags = ImmutableMap.of(Constants.Metrics.Tag.PLUGIN_NAME, entry.getKey().getFirst(),
+                                                   Constants.Metrics.Tag.PLUGIN_TYPE, entry.getKey().getSecond());
+        metricsCollectionService.getContext(tags).gauge(Constants.Metrics.Program.PLUGIN_COUNT, entry.getValue());
+      }
+
+      if (deletedPlugins == null) {
+        return;
+      }
+
+      // emit 0 metric for deleted plugins
+      for (ImmutablePair<String, String> pair: Sets.difference(deletedPlugins, Sets.newHashSet(remainingPlugins))) {
```

Comment:
I'm not 100% sure what this deleted plugins contains but is it possible there's a collision between this and the remaining plugins? 

Ex. I delete a pipeline with GCS source but another pipeline also has a GCS source, the count should be 1, not 0 right?